src/components/victory-axis.jsx

```diff
@@ -291,22 +294,21 @@ export default class VictoryAxis extends React.Component {
   }
 
   getTicks(props) {
-    let t;
-    if (this.stringMap) {
-      t = _.values(this.stringMap);
-    } else if (props.tickValues) {
-      t = props.tickValues;
+    if (props.tickValues) {
+      // Since we declared that `tickValues` must be a homogenous array, we only
+      // need to do a string check on the first item.
+      if (typeof props.tickValues[0] === "string") {
+        return _.range(1, props.tickValues.length + 1);
+      }
+      return props.tickValues;
     } else if (_.isFunction(this.scale.ticks)) {
       const ticks = this.scale.ticks(props.tickCount);
       if (props.crossAxis) {
-        t = _.includes(ticks, 0) ? _.without(ticks, 0) : ticks;
-      } else {
-        t = ticks;
+        return _.includes(ticks, 0) ? _.without(ticks, 0) : ticks;
       }
-    } else {
-      t = this.scale.domain();
+      return ticks;
     }
-    return Array.isArray(t) ? t : [t];
+    return this.scale.domain();
```

Comment:
This `isArray` didn't seem to be doing anything, as I think all the `if` branches above returned arrays anyway. @boygirl Correct me if that's wrong!